packages/clerk-js/src/utils/queryStateParams.ts

```diff
@@ -0,0 +1,38 @@
+import { getClerkQueryParam, removeClerkQueryParam } from '../utils';
+
+export const decodeBase64Json = (value: string) => {
+  return JSON.parse(atob(value));
+};
+
+export const encodeBase64Json = (value: { path: string; componentName: string }) => {
+  return btoa(JSON.stringify(value));
+};
+
+export const readAndRemoveStateParam = () => {
+  const urlClerkState = getClerkQueryParam('__clerk_state') ?? '';
+  removeClerkQueryParam('__clerk_state');
+  return urlClerkState ? decodeBase64Json(urlClerkState) : null;
+};
+
+type SerializeAndAppendModalStateProps = { url: string; isModal: boolean; currentPath: string; componentName: string };
+export const serializeAndAppendModalState = ({
+  url,
+  isModal,
+  currentPath,
+  componentName,
+}: SerializeAndAppendModalStateProps) => {
+  if (!isModal) {
+    return url;
+  }
```

Comment:
This has me a bit confused. Why are we passing `isModal` here if we are just going to return the `url`? Shouldn't this be handled outside the function? Am I missing something?

Reply:
@desiprisg I refactored the function as you suggested.